src/main/java/org/openstreetmap/atlas/utilities/collections/StreamIterable.java

```diff
@@ -30,6 +30,30 @@ protected StreamIterable(final Iterable<T> source)
         this.source = source;
     }
 
+    /**
+     * Test whether all elements from iterable matches given predicate or not
+     *
+     * @param predicate
+     *            Predicate to test
+     * @return True when given predicate is true for all entities in iterable, else false
+     */
+    public boolean allMatch(final Predicate<T> predicate)
+    {
+        return StreamSupport.stream(this.source.spliterator(), false).allMatch(predicate);
+    }
+
+    /**
+     * Test whether any of the elements from iterable matches given predicate
+     *
+     * @param predicate
+     *            Predicate to test
+     * @return True when given predicate is true for any one entity in iterable, else false
+     */
+    public boolean anyMatch(final Predicate<T> predicate)
+    {
+        return StreamSupport.stream(this.source.spliterator(), false).anyMatch(predicate);
```

Comment:
Any reason not to use parallel stream here?
```
return StreamSupport.stream(this.source.spliterator(), true).allMatch(predicate)
```

Reply:
See above